src/RouteMapOverlay.cpp

```diff
@@ -1686,35 +1686,30 @@ Position* RouteMapOverlay::getClosestRoutePositionFromCursor(double cursorLat, d
      */
     
     double dist = INFINITY;
-    Position *pos = last_destination_position;
-    Position *newPos = nullptr;
-    
-    // Get position first
-    for(Position *p = pos; p && p->parent; p = p->parent)
+    std::list<PlotData> plot = GetPlotData(false);
+    bool found = false;
+    for ( const auto &it : plot)
     {
         // Calculate distance
         // Almost like a plan (x,y) because of small distance -- is that correct?
```

Comment:
Is this an optimization?  Why not rebuild the plot data rather than compute the position?